django/db/models/utils.py

```diff
@@ -19,3 +22,16 @@ def make_model_tuple(model):
             "Invalid model reference '%s'. String model references "
             "must be of the form 'app_label.ModelName'." % model
         )
+
+
+def get_field_from_path(model, path):
+    pieces = path.split(LOOKUP_SEP)
+    fields = []
+    for piece in pieces:
+        parent = model
+        if (fields and fields[-1].is_relation and
```

Comment:
There must be a clearer way to spell out the intent of this block. 

Thoughts: 

* Using `all()` with a comment for why for each element? (We can tidy that once it's clearer)
* Giving this `fields[-1].remote_field.model` a meaningful name would help.